src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -316,7 +316,9 @@ public List<Feedback> copyFeedbackToNewResult(Result newResult, Result oldResult
     private void copyFeedbackToResult(Result result, List<Feedback> feedbacks) {
         feedbacks.forEach(feedback -> {
             Feedback newFeedback = feedback.copyFeedback();
-            newFeedback.setPositiveViaCredits();
+            if (newFeedback.isPositive() == null) {
+                newFeedback.setPositiveViaCredits();
+            }
```

Comment:
Do we actually need this in this place or can we integrate it inside of copyFeedback?

Reply:
I agree, a boolean being `null` is IMO something that should be hidden / fixed in the background